core/src/main/java/hudson/model/Run.java

```diff
@@ -1441,7 +1449,9 @@ public String toString() {
 
     /**
      * Returns an input stream that reads from the log file.
-     * It will use a gzip-compressed log file (log.gz) if that exists.
+     *
+     * It will use a gzip-compressed log file (log.gz) if that exists, decompressing
+     * it on the fly to return the uncompressed data.
```

Comment:
Not a fan of this being documented in the API (strictly speaking, this means JEP-210 is invalid), but at least this PR doesn't add it.

Reply:
Yeah to complement #3963 I would suggest simply deleting this sentence. How or where the log text is stored is a matter of no concern to the caller of this method, and for Pipeline there is no built-in gzip support (this implementation is not used).

Reply:
+1, also this would become wrong indeed if we ended up merging https://github.com/jenkinsci/jenkins/pull/3966

Reply:
Removing it at all since there is a consensus. Not much related to the PR anyway